src/main/scala/firrtl/transforms/ConstantPropagation.scala

```diff
@@ -414,7 +418,7 @@ class ConstantPropagation extends Transform with ResolvedAnnotationPaths {
           constOutputs(pname) = paddedLit
         // Const prop registers that are driven by a mux tree containing only instances of one constant or self-assigns
         // This requires that reset has been made explicit
-        case Connect(_, lref @ WRef(lname, ltpe, RegKind, _), rhs) if !dontTouches.contains(lname) =>
+        case Connect(_, lref @ WRef(lname, ltpe, RegKind, _), rhs) if !dontTouches(lname) && !asyncResetRegs(lname) =>
```

Comment:
Is there a reason you aren't checking dontTouches.contains anymore?

Reply:
For sets, `set.contains(blah)` is the exact same as `set(blah)`, I just changed it to make it fit on one line better 😬

Reply:
Wow. TIL.